src/sulogin.c

```diff
@@ -202,10 +203,10 @@ static void catch_signals (unused int sig)
 			execl (PATH_TELINIT, "telinit", RUNLEVEL, (char *) 0);
 #endif
 			exit (0);
-		} else {
-			STRFCPY (pass, cp);
-			strzero (cp);
 		}
+		STRFCPY (pass, cp);
```

Comment:
What would happen if `cp` is NULL?

Reply:
That you'd have entered in the above conditional, and if `execl(3)` is successful, you'd have changed the current process into `telinit`, so this code would not be executed.  If that call failed, it would return -1 (correctly ignored), and then we'd have run `exit(EXIT_SUCCESS)`, which would have immediately quitted the program, and also the code would not be executed.

I question why `exit(3)` is being passed `0` instead of `1`, but I didn't care too much to investigate.

But it's clear that this code is not being executed if `cp == NULL`.

Reply:
You are right. Thanks for clarifying.

Reply:
You're welcome to ask :)